torch/_torch_docs.py

```diff
@@ -10730,12 +10730,165 @@ def merge_dicts(*dicts):
+    >>> torch.trapezoid(y, x)
+    tensor([25., 65.])
+
+    >>> # Compute the integral of y with 2-D tensors of coordinates,
+    >>> # but use dim=0 to integrate over the outer dimension.
+    >>> y = torch.tensor([[0, 10, 20], [10, 20, 40]])
+    >>> x = torch.tensor([[0, 0, 0], [1, 1, 1]])
+    >>> torch.trapezoid(y, x, dim=0)
+    tensor([0., 15., 30])
+
+    >>> # Since the dimension of two tensors do not match, it
+    >>> # attempts to broadcast x to match the dimension of y.
+    >>> y = torch.tensor([[10, 10, 20], [0, 10, 20]])
+    >>> x = torch.tensor([0, 1, 2])
+    >>> torch.trapezoid(y, x)
+    tensor([25., 20.])
+
+    >>> # Compute the integral of y with 3-D tensors of coordinates.
+    >>> y = torch.tensor(np.arange(8).reshape((2,2,2)))
+    >>> y
+    tensor([[[0, 1],
+             [2, 3]],
+            [[4, 5],
+             [6, 7]]])
+    >>> x = torch.tensor(np.arange(8).reshape((2,2,2)))
+    >>> x
+    tensor([[[0, 1],
+             [2, 3]],
+            [[4, 5],
+             [6, 7]]])
+    >>> torch.trapezoid(y, x)
+    tensor([[0.5000, 2.5000],
+            [4.5000, 6.5000]])
+
+.. function:: trapezoid(y, *, dx=1, dim=-1) -> Tensor
+
+Alternatively, instead of specifying all values to integrate over with tensor :attr:`x`,
+you can pass in a scalar argument :attr:`dx`, which represents the uniform distance
+between the sample points. By default, :attr:`dx=1`, that is equivalent to
+having :math:`x = [0.0, 1.0, ...]` up to the number of elements in :attr:`y`.
+
+Note that you cannot pass both :attr:`x` and :attr:`dx` as arguments, because :attr:`x` allows you
+to specify the desired spacing for every interval within the integral, such that
+passing in :attr:`dx` will not provide additional useful information.
+
+
+Arguments:
+    y (Tensor): The values of the function to integrate
+
+Keyword args:
+    dx (Scalar): The distance between points at which `y` is sampled.
+        Currently, only float/double types are supported.
+    dim (int): The dimension along which to integrate.
+        By default, use the last (inner-most) dimension.
+
+Returns:
+    A Tensor with the same shape as the input, except with `dim` removed.
+    Each element of the returned tensor represents the estimated integral
+    :math:`\int y\,dx` along :attr:`dim`.
+
+Examples::
+
+    >>> # By default, dx = 1.0.
+    >>> y = torch.tensor([0, 10, 20])
+    >>> torch.trapezoid(y)
+    tensor(20.)
+    >>> torch.trapezoid(y, dx=2.0)
+    tensor(40.)
+
+    >>> # The same dx (spacing) is used for each integration.
+    >>> y = torch.tensor([[0, 10, 20], [0, 20, 40]])
+    >>> torch.trapezoid(y, dx=2.0)
+    tensor([40., 80.])
+
+    >>> # You can also specify dim while using dx.
+    >>> y = torch.tensor([[0, 10, 20], [0, 20, 40]])
+    >>> torch.trapezoid(y, dx=2.0, dim=0)
+    tensor([40., 30., 60.])
 """)
 
 add_docstr(torch.trapz,
            r"""
 trapz(y, x, *, dim=-1) -> Tensor
 
+This is deprecated, please use :func:`torch.trapezoid`.
```

Comment:
Let's give this the same signature as trapezoid and just remove all documentation except:

"Alias for :func:\`torch.trapezoid.\`."

The documentation in _tensor_docs.py will also have to be updated with the correct signature, but it's OK for the method documentation to say it's the method version of `trapz`, it doesn't need to explain it's an alias.

Reply:
Okay, I will change the documentation to "Alias for :func:`torch.trapezoid.`."

Let me know if you want me to add wording that indicate the function is "deprecated".